webapps/frontend/app/viewmodels/site/updates.js

```diff
@@ -15,265 +15,322 @@
+                                        return package1.name < package2.name ? -1 : 1;
+                                    });
+                                });
+                            });
+                            // Sort nodes by IP
+                            self.storageNodes.sort(function(sn1, sn2) {
+                                return sn1().ip < sn2().ip ? -1 : 1;
+                            });
+                            deferred.resolve();
+                        })
+                        .fail(deferred.reject)
+                        .always(function() {
+                            if (self.loadedInfo() === false) {
+                                self.loadedInfo(true);
+                            }
+                        })
+                }
             }).promise();
         };
-
-        self.updateVolumedriver = function() {
-            if (self.upgradeOngoing() === true) {  // Checked by presence of file /etc/upgrade_ongoing on any of the storagerouters
-                return;
+        self.refresh = function() {
+            self.refreshing(true);
+            if (generic.xhrCompleted(self.refreshPackageInfoHandle)) {
+                self.refreshPackageInfoHandle = api.get('storagerouters/' + self.storageRouters()[0].guid() + '/refresh_package_information')
+                    .then(shared.tasks.wait)
+                    .done(function () {
+                        generic.alertSuccess(
+                            $.t('ovs:updates.refresh.success'),
+                            $.t('ovs:updates.refresh.success_msg')
+                        );
+                    })
+                    .fail(function (error) {
+                        error = generic.extractErrorMessage(error);
+                        generic.alertError(
+                            $.t('ovs:generic.error'),
+                            $.t('ovs:updates.refresh.failure_msg', {error: error})
+                        );
+                    })
+                    .always(function () {
+                        self.refreshing(false);
+                    });
+                generic.alertInfo(
+                    $.t('ovs:updates.refresh.started'),
+                    $.t('ovs:updates.refresh.started_msg')
+                );
             }
-            self.upgradeOngoing(true);
-            return $.Deferred(function(deferred) {
-                var downtimes = [];
-                var prerequisites = [];
-                $.each(self.updates().volumedriver.downtime, function(index, downtime) {
-                    if (downtime[2] === null) {
-                        downtimes.push($.t(downtime[0] + ':downtime.' + downtime[1]))
-                    } else {
-                        downtimes.push($.t(downtime[0] + ':downtime.' + downtime[1]) + ': ' + downtime[2])
-                    }
+        };
+        self.expandCollapseAll = function(value) {
+            $.each(self.storageRouters(), function(index, sr) {
```

Comment:
You also could just make this one a observable which you can set to ```true``` or ```false```, and then in the individual collapsable parts check for its own observable, and all its parents. This also means that if you collapse and expand, all internal collapsion states will be identical.